### PR TITLE
chore: Prepare for loose publication using antora.yml from the branch

### PR DESCRIPTION
**Pull request by themr0c**
### What does this PR do?

Prepare for loose publication using antora.yml from the branch and antora-playbook.yml from master

Enables to:

* Merge PR for next version to the master branch.

New publication flow:

* Use the `antora-playbook-publication.yml` located in master branch to orchestrate publication. See https://ci.eclipse.org/che/job/che-docs-pipeline/job/master/.
* Use the content from the stable branch referenced in `antora-playbook-publication.yml`, currently `7.31.x `, as a source to publish the content.

New requirements:

* All AsciiDoc attributes are defined in the component definition file `antora.yml` located in the publication branch defined in `antora-playbook-publication.yml`, currently `7.31.x `.
* Scripts search for the value of the `prod-short` attribute in `antora.yml`
* It makes no sense for publication to run `./tools/environment_docs_gen.sh` and   `./tools/checluster_docs_gen.sh` from master branch. 
* We need to run `./tools/environment_docs_gen.sh` and   `./tools/checluster_docs_gen.sh` in the context of the release branch, currently `7.31.x `.

Known limitations:

* Run `./tools/environment_docs_gen.sh` and     `./tools/checluster_docs_gen.sh` when creating/updating the stable branch (currently `7.31.x` ). Automate in `make-release.sh`? 

* Version is set to next (`7.32.x`) in master branch =>  failures for `./tools/environment_docs_gen.sh` and     `./tools/checluster_docs_gen.sh`

### What issues does this PR fix or reference?

https://issues.redhat.com/browse/RHDEVDOCS-2810

### PR Checklist

As the author of this Pull Request I made sure that:

- [x] `vale` has been run successfully against the PR branch
- [x] Link checker has been run successfully against the PR branch
- [x] Documentation describes a scenario that is already covered by QE tests, otherwise an issue has been created and acknowledged by Che QE team
- [x] Changed article references are updated where they are used (or a redirect has been configured on the docs side):
    - [x] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/src/services/bootstrap/branding.constant.ts)
    - [x] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/master/src/constants.ts)

### CODE REVIEW

**Review by benoitf — APPROVED** [review]
LGTM

**Review by MichalMaler — APPROVED** [review]

**Comment by themr0c** [comment]
Now we have an issue using master for an unreleased version: the documentation then points to files which are not existing yet:

https://github.com/eclipse-che/che-plugin-registry/tree/7.32.0
https://raw.githubusercontent.com/eclipse/che-theia/7.32.0/extensions/eclipse-che-theia-plugin/src/che.d.ts

**Comment by benoitf** [comment]
we could use in links `HEAD` as variable value when it's in main branch

https://github.com/eclipse-che/che-plugin-registry/tree/HEAD
https://raw.githubusercontent.com/eclipse/che-theia/HEAD/extensions/eclipse-che-theia-plugin/src/che.d.ts

**Comment by themr0c** [comment]
Don't merge until the  issue pointed there has a proper resolution: https://github.com/eclipse/che-docs/pull/2022#issuecomment-859371682

Suggestion by @benoitf is nice but difficult to implement: we rely on the prod-ver to set the coordinates of the links. Converting to "use prod-ver here when it is released, HEAD when it is not released" is not something trivial to implement in AsciiDoc. We may end-up with another script to handle the situation, but I would like to avoid it if we can find something simpler.

**Comment by themr0c** [comment]
In current state it breaks the logic of  `make-release.sh`.

2 main changes:

* We need to run the generation scripts at release time:
  * `./tools/checluster_docs_gen.sh` 
  * `./tools/environment_docs_gen.sh`
* Different handling of versions

What a release should look like:

* Invoke the release script with the version to release: `make-release.sh --release_version $RELEASE_VERSION --next $NEXT_VERSION`

### Validate the $RELEASE_VERSION 

* Validate that $RELEASE_VERSION  is major.minor.patch
* Define RELEASE_BRANCH as major.minor.x

###  Prepare the release branch

* In `antora.yaml`, set  `prod-ver` to ${RELEASE_VERSION}, and other attributes accordingly.
* Generate docs for ${RELEASE_VERSION}: 
  * `./tools/checluster_docs_gen.sh` 
  * `./tools/environment_docs_gen.sh`
* Commit and push the results of the change in the `${RELEASE_BRANCH}` branch
* Tag for `${RELEASE_VERSION}`


###  Prepare the PR for the main branch (`master`)

* In `antora.yaml`, set `prod-ver` to ${NEXT_VERSION}, and other attributes accordingly.

**Comment by themr0c** [comment]
@mkuznyetsov @nickboldt As this pull request is introducing significant changes in make-release.sh, I wish you could review it before merge.

Summarizing the changes:

* For development: we build HEAD, using the antora-playbook-for-development.yml 

* For publication: we build the latest MAJOR.MINOR.x branch, using the antora-playbook-for-publication.yml located in the main branch

* To release a new version:
  * We update the version attributes in antora.yml in the `MAJOR.MINOR.x` branch
  * We update the reference to the  `MAJOR.MINOR.x` branch in the antora-playbook-for-publication.yml in main branch

To test, I am using the following command:

```
./make-release.sh -v 7.35.1  -tmp --nopush
```
NB: I have a known chicken and egg problem with one YAML file not yet present in master.

**Comment by themr0c** [comment]
```
./make-release.sh -v 7.35.1  -tmp --nopush                 127 
[INFO] Version format for: 7.35.1 is in form MAJOR.MINOR.PATCH.
Cloning into 'tmp-make-release.sh-7.35.1'...
remote: Enumerating objects: 34141, done.
remote: Counting objects: 100% (298/298), done.
remote: Compressing objects: 100% (207/207), done.
remote: Total 34141 (delta 128), reused 171 (delta 90), pack-reused 33843
Receiving objects: 100% (34141/34141), 150.32 MiB | 762.00 KiB/s, done.
Resolving deltas: 100% (22230/22230), done.
Already up to date.
[INFO] Updating branch: 7.35.x from branch: master.
Branch '7.35.x' set up to track remote branch '7.35.x' from 'origin'.
Switched to a new branch '7.35.x'
From github.com:eclipse/che-docs
 * branch              master     -> FETCH_HEAD
dropping 2bf4e83eda55e6b32ba9cc2e72676db5c5c9ae81 chore: Bump to 7.35.0 in 7.35.x -- patch contents already upstream
Successfully rebased and updated refs/heads/7.35.x.
[INFO] Updating to version: 7.35.1 is an increment for current version: 7.35.0.
[INFO] Updating file: antora.yml on branch: 7.35.x: setting attribute: prod-ver-major: "7"
[INFO] Updating file: antora.yml on branch: 7.35.x: setting attribute: prod-ver: "7.35"
[INFO] Updating file: antora.yml on branch: 7.35.x: setting attribute: prod-ver-patch: "7.35.1"
[INFO] Updating file: antora.yml on branch: 7.35.x: setting attribute: prod-prev-ver: "7.34"
Trying to read current product version from /tmp/tmp-make-release.sh-7.35.1/antora-playbook.yml...
Detected version: 7.35.x
Trying to read product name from /tmp/tmp-make-release.sh-7.35.1/antora-playbook.yml...
Detected product: che
Fetching property files content from GitHub...
Fetching content done. CRD version: apiextensions.k8s.io/v1. Trying to parse it.
Parsing section: server
Parsing section: database
Parsing section: auth
Parsing section: storage
Parsing section: k8s
Parsing section: metrics
Parsing section: status
Processing done. Output file is /tmp/tmp-make-release.sh-7.35.1/modules/installation-guide/examples/checluster-properties.adoc
Trying to read current product version from /tmp/tmp-make-release.sh-7.35.1/antora-playbook.yml...
Detected version: 7.35.x
Fetching property files content from GitHub...
Fetching content done. Trying to parse it.
   Found begin of topic: {prod-short} server
   Found begin of topic: Authentication parameters
   Found begin of topic: Internal
   Found begin of topic: Kubernetes Infra parameters
   Found begin of topic: OpenShift Infra parameters
   Found begin of topic: Experimental properties
   Found begin of topic: Configuration of the major WebSocket endpoint
   Found begin of topic: CORS settings
   Found begin of topic: Factory defaults
   Found begin of topic: Devfile defaults
   Found begin of topic: Che system
   Found begin of topic: Workspace limits
   Found begin of topic: Users workspace limits
   Found begin of topic: Organizations workspace limits
   Found begin of topic: Multi-user-specific OpenShift infrastructure configuration
   Found begin of topic: Keycloak configuration
Processing done. Output file is /tmp/tmp-make-release.sh-7.35.1/modules/installation-guide/examples/system-variables.adoc
[7.35.x 71a58380] release: Bump version to 7.35.1
 3 files changed, 60 insertions(+), 43 deletions(-)
[INFO] Creating new branch: release-7.35.1 from branch: master.
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
Switched to a new branch 'release-7.35.1'
[INFO] Updating file: antora-playbook-for-publication.yml on branch: release-7.35.1: setting attribute: branches: "7.35.x"
sed: can't read antora-playbook-for-publication.yml: No such file or directory
```

File: antora-playbook-for-publication.yml
Comment (nickboldt) [inline antora-playbook-for-publication.yml]: 7.32.x? that's oooooold.

File: antora-playbook-for-publication.yml
Comment (nickboldt) [inline antora-playbook-for-publication.yml]: 7.31? that's super oooooold. and not the same as line 14, either. (?)

File: antora.yml
Comment (nickboldt) [inline antora.yml]: this is minimum ocp version, I assume? we also support 4.8 (and 2.12 will likely support 4.9)

File: make-release.sh
Comment (nickboldt) [inline make-release.sh:35]: we don't want to fail on error?

File: make-release.sh
Comment (nickboldt) [inline make-release.sh:90]: do we need to pull if we've just cloned?

File: make-release.sh
Comment (nickboldt) [inline make-release.sh:146]: any support for if tag already exists?

File: make-release.sh
Comment (nickboldt) [inline make-release.sh]: This could be yq instead of sed but for simply one-line changes I guess this is just as easy :D

File: modules/contributor-guide/partials/ref_che-theia-plug-in-api.adoc
Comment (nickboldt) [inline modules/contributor-guide/partials/ref_che-theia-plug-in-api.adoc:43]: there's nothing in https://github.com/eclipse-che/che-theia/blob/main/extensions/eclipse-che-theia-plugin/src/che.d.ts anymore. Perhaps this list of resources has moved? @benoitf

File: modules/end-user-guide/partials/ref_che-theia-plug-in-metadata.adoc
Comment (nickboldt) [inline modules/end-user-guide/partials/ref_che-theia-plug-in-metadata.adoc]: the `che-theia-plugins.yaml` contains
should be either
`che-theia-plugins.yaml` contains
or
the `che-theia-plugins.yaml` file contains

File: tools/publication.sh
Comment (nickboldt) [inline tools/publication.sh:1]: why `env sh` and not `bash`  like other scripts?

File: tools/publication.sh
Comment (nickboldt) [inline tools/publication.sh:12]: in runnerpublication.sh you have a comment `# Fail on errors and display commands` ... should it be here too?

File: make-release.sh
Comment (nickboldt) [inline make-release.sh]: why `bash -c tools/...` ? why not just `./tools/...` ?

**Review by nickboldt — CHANGES_REQUESTED** [review]
some questions, some comments, some suggestions. most are safe to ignore, but the versions in antora-playbook-for-publication.yml looks weird. 

Also the one question for Florent should be addressed.

File: antora-playbook-for-publication.yml
Comment (themr0c) [reply antora-playbook-for-publication.yml]:
```suggestion
      branches: 7.35.x
```

File: antora-playbook-for-publication.yml
Comment (themr0c) [reply antora-playbook-for-publication.yml]:
```suggestion
      edit_url: "https://github.com/eclipse/che-docs/edit/7.35.x/{path}"
```

File: antora.yml
Comment (themr0c) [reply antora.yml]: This attribute is never used. Removing it.
```suggestion
```

File: make-release.sh
Comment (themr0c) [reply make-release.sh:35]: Yes, the statement has been moved to line 25.

File: make-release.sh
Comment (themr0c) [reply make-release.sh:90]: To face the situation when a merge occured during clone. (It happened during development, because I was using the clone wait time to finish reviews...)

File: make-release.sh
Comment (themr0c) [reply make-release.sh:146]: Then the release already exists, and we should already have failed when checking if version is incremented.

File: make-release.sh
Comment (themr0c) [reply make-release.sh]: Interesting comment, I had never thought about using yq to modify YAML.

File: modules/contributor-guide/partials/ref_che-theia-plug-in-api.adoc
Comment (themr0c) [inline modules/contributor-guide/partials/ref_che-theia-plug-in-api.adoc]: Removing reference to https://github.com/eclipse-che/che-theia/blob/main/extensions/eclipse-che-theia-plugin/src/che.d.ts as it is now empty.

```suggestion
```

File: modules/contributor-guide/partials/ref_che-theia-plug-in-api.adoc
Comment (themr0c) [reply modules/contributor-guide/partials/ref_che-theia-plug-in-api.adoc:43]: Removing reference to https://github.com/eclipse-che/che-theia/blob/main/extensions/eclipse-che-theia-plugin/src/che.d.ts as it is now empty.

File: make-release.sh
Comment (themr0c) [reply make-release.sh]: No good reason. Fixed.

File: tools/publication.sh
Comment (themr0c) [reply tools/publication.sh:1]: Nice spot, `#!/usr/bin/env sh` is what we have most. `get_version_next.sh` and `get_version_released.sh`are leftovers from development time that I should not have committed.

File: tools/publication.sh
Comment (themr0c) [reply tools/publication.sh:12]: done

File: modules/end-user-guide/partials/ref_che-theia-plug-in-metadata.adoc
Comment (themr0c) [reply modules/end-user-guide/partials/ref_che-theia-plug-in-metadata.adoc]:
```suggestion
In the link:https://github.com/eclipse-che/che-plugin-registry[{prod-short} plug-ins registry repository], the `che-theia-plugins.yaml` file contains a list of all Che-Theia plug-ins in the registry. Build the registry container image to generate the `meta.yaml` file.
```

**Comment by themr0c** [comment]
@nickboldt addressed all comments.

* Changing the version in antora.yml asciidoc attributes in both the version branch and main branch
* Changing the version branch in antora-playbook-for-publication.yml in both the version branch and main branch
* Consistency: variables in upper case, functions in camelCase.

**Review by nickboldt — APPROVED** [review]
leems segit. :D

**Pull request merged**
